### DISCUSSION TITLE
Problem while using slither

### DISCUSSION THREAD

**Discussion by patilchirag99** [crytic/slither#1939] [Installation]
slither slither .
pygame 2.4.0 (SDL 2.26.4, Python 3.10.10)
Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/bin/slither", line 5, in <module>
    from slither.__main__ import main
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/slither/__main__.py", line 26, in <module>
    from slither.slither import Slither
ImportError: cannot import name 'Slither' from 'slither.slither' (/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/slither/slither.py)

Getting above error while using slither on Mac

**Comment by 0xalpharush** [comment]
Hi you've installed a different package (https://pypi.org/project/slither/) and not slither-analyzer (https://pypi.org/project/slither-analyzer/). You can fix this by running `pip uninstall slither && pip install slither-analyzer`.

**Reply by patilchirag99** [reply]
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
eth-bloom 1.0.4 requires eth-hash[pycryptodome]<0.4.0,>=0.3.1, but you have eth-hash 0.5.1 which is incompatible.
mythril 0.23.22 requires eth-abi<3.0.0,>=2.0.0b4, but you have eth-abi 4.0.0 which is incompatible.
mythril 0.23.22 requires eth-account<0.6.0,>=0.5.6, but you have eth-account 0.8.0 which is incompatible.
mythril 0.23.22 requires eth-hash<0.4.0,>=0.3.1, but you have eth-hash 0.5.1 which is incompatible.
mythril 0.23.22 requires eth-keyfile<0.6.0,>=0.5.1, but you have eth-keyfile 0.6.1 which is incompatible.
mythril 0.23.22 requires eth-keys<0.4.0,>=0.2.1, but you have eth-keys 0.4.0 which is incompatible.
mythril 0.23.22 requires eth-rlp<0.3.0,>=0.1.0, but you have eth-rlp 0.3.0 which is incompatible.
mythril 0.23.22 requires eth-typing<3.0.0,>=2.1.0, but you have eth-typing 3.3.0 which is incompatible.
mythril 0.23.22 requires eth-utils<2, but you have eth-utils 2.1.0 which is incompatible.
mythril 0.23.22 requires MarkupSafe<2.1.0, but you have markupsafe 2.1.2 which is incompatible.
mythril 0.23.22 requires rlp<3, but you have rlp 3.0.0 which is incompatible.
mythril 0.23.22 requires typing-extensions<4,>=3.7.4, but you have typing-extensions 4.6.2 which is incompatible.
py-ecc 4.1.0 requires eth-typing<3.0.0,>=2.1.0, but you have eth-typing 3.3.0 which is incompatible.
py-ecc 4.1.0 requires eth-utils<2,>=1.3.0, but you have eth-utils 2.1.0 which is incompatible.
py-evm 0.5.0a1 requires eth-keys<0.4.0,>=0.2.1, but you have eth-keys 0.4.0 which is incompatible.
py-evm 0.5.0a1 requires eth-typing<3.0.0,>=2.2.0, but you have eth-typing 3.3.0 which is incompatible.
py-evm 0.5.0a1 requires eth-utils<2.0.0,>=1.9.4, but you have eth-utils 2.1.0 which is incompatible.
py-evm 0.5.0a1 requires rlp<3,>=2, but you have rlp 3.0.0 which is incompatible.
trie 2.0.0a5 requires eth-utils<2.0.0,>=1.6.1, but you have eth-utils 2.1.0 which is incompatible.
trie 2.0.0a5 requires rlp<3,>=1, but you have rlp 3.0.0 which is incompatible.
trie 2.0.0a5 requires typing-extensions<4,>=3.7.4, but you have typing-extensions 4.6.2 which is incompatible.

Getting this error now. Kindly help

**Reply by elopez** [reply]
Hi! it looks like you've got incompatible packages installed. Uninstall everything that's mentioned to clear out the conflict and give it another go:

```
pip3 uninstall slither-analyzer crytic-compile solc-select slither eth-bloom mythril py-ecc py-evm trie eth-abi eth-account eth-hash eth-keyfile eth-keys eth-rlp eth-typing eth-utils markupsafe rlp typing-extensions

pip3 install slither-analyzer
```

Alternatively, look into using Python virtualenvs, or something like `pipx`. They let you install packages in isolation so these kinds of problems don't occur.
